crates/iceberg/src/scan.rs

```diff
@@ -187,6 +190,22 @@ impl TableScan {
         let mut arrow_reader_builder =
             ArrowReaderBuilder::new(self.file_io.clone(), self.schema.clone());
 
+        let mut field_ids = vec![];
+        for column_name in &self.column_names {
+            let field_id = self.schema.field_id_by_name(column_name).ok_or_else(|| {
```

Comment:
As discussed in #244, we need to do two checks here to ensure that it's valid:

1. The field is a direct child of schema, e.g. not a nested field. We can do this by calling `Schema::as_struct::field_by_id::is_some`
2. Ensure that this field is primitive type.

Reply:
Added these checks.